drivers/ethernet/Kconfig.stm32_hal

```diff
@@ -22,13 +22,13 @@ choice ETH_STM32_HAL_API_VERSION
 
 config ETH_STM32_HAL_API_V2
 	bool "Use new HAL driver"
-	depends on SOC_SERIES_STM32H7X || SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X
+	depends on SOC_SERIES_STM32H7X || SOC_SERIES_STM32H5X || SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X
 	help
 	  Use the new HAL driver instead of the legacy one.
 
 config ETH_STM32_HAL_API_V1
 	bool "Use new HAL driver"
-	select DEPRECATED if SOC_SERIES_STM32H7X || SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X
+	select DEPRECATED if SOC_SERIES_STM32H7X || SOC_SERIES_STM32H5X || SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X
```

Comment:
`ETH_STM32_HAL_API_V1` is not supported on H5, so updating this line is not required and instead we can add a:
```
depends on !SOC_SERIES_STM32H5X
```

Reply:
OK